tests/Owin.Security.OpenIdConnect.Server.Tests/OAuth2AuthorizationClientCredentialsGrantTests.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Protocols;
+using Microsoft.Owin.Testing;
+using Xunit;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests {
+    public class OAuth2AuthorizationClientCredentialsGrantTests {
+        [Fact]
+        public async Task ClientCredentials_MissingCredentials_BadRequest() {
+            var server = TestServer.Create(app => { app.UseOpenIdConnectServer(new OpenIdConnectServerOptions { AllowInsecureHttp = true }); });
+
+            var openIdConnectMessage = new OpenIdConnectMessage {
+                RequestType = OpenIdConnectRequestType.TokenRequest,
+                GrantType = "client_credentials",
+                TokenEndpoint = "connect/token"
+            };
+
+            var response = await server.SendMessageAsync(openIdConnectMessage);
+            var openIdResponseMessage = await response.Content.ReadAsOpenIdConnectMessageAsync();
+
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            Assert.Equal("invalid_client", openIdResponseMessage.Error);
```

Comment:
Feel free to add `InternalsVisibleToAttribute` to the main project if you want to reuse the internal constants.